File: shared/src/main/scala/stasis/shared/api/requests/CreateDatasetDefinition.scala
package stasis.shared.api.requests

import stasis.shared.model.datasets.DatasetDefinition
import stasis.shared.model.devices.Device

final case class CreateDatasetDefinition(
  info: String,
  device: Device.Id,
  redundantCopies: Int,
  existingVersions: DatasetDefinition.Retention,
  removedVersions: DatasetDefinition.Retention
)

object CreateDatasetDefinition {
  implicit class RequestToDefinition(request: CreateDatasetDefinition) {
    def toDefinition: DatasetDefinition =
      DatasetDefinition(
        id = DatasetDefinition.generateId(),
        info = request.info,
        device = request.device,
        redundantCopies = request.redundantCopies,
        existingVersions = request.existingVersions,
        removedVersions = request.removedVersions
      )
  }
}

File: client-android/lib/src/main/kotlin/stasis/client_android/lib/model/server/api/requests/CreateDatasetDefinition.kt
package stasis.client_android.lib.model.server.api.requests

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import stasis.client_android.lib.model.server.datasets.DatasetDefinition
import stasis.client_android.lib.model.server.devices.DeviceId

@JsonClass(generateAdapter = true)
data class CreateDatasetDefinition(
    val info: String,
    val device: DeviceId,
    @field:Json(name = "redundant_copies")
    val redundantCopies: Int,
    @field:Json(name = "existing_versions")
    val existingVersions: DatasetDefinition.Retention,
    @field:Json(name = "removed_versions")
    val removedVersions: DatasetDefinition.Retention
)
